### PR TITLE
No core-js in core

### LINKED ISSUE #1601: upgrade core-js to 3.x
### Prerequisites

- [ ] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/);
- [ ] In the case of a bug report, I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [ ] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net/n1k0/f2y3fq7L/6/) or a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### Description

[Description of the bug or feature]

### Steps to Reproduce

1. [First Step]
2. [Second Step]
3. [and so on...]

#### Expected behavior

[What you expected to happen]

#### Actual behavior

[What actually happened]

### Version

You can usually get this information in your `package.json` or in the file URL if you're using the unpkg one.

### PR DESCRIPTION
**Pull request by wegry**
### Reasons for making this change

Two versions of core-js currently exist in my app because @rjsf/core pulls in @babel/runtime currently. There are two direct imports of core-js that I've opted to pull in pure imports for.

For the rationale of why library code shouldn't include polyfills, this comment
https://github.com/w3ctag/polyfills/issues/6#issuecomment-272647475 explains my rationale. This is a breaking change though, if downstream folks were depending on `@babel/runtime` being around.

If this is related to existing tickets, include links to them as well. fixes #1601

Recreates PR for #2049

### Checklist

* [x] **I'm updating documentation**
  - [x] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [X] **I'm adding or updating code**
  - [ ] I've added and/or updated tests
  - [x] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
* [x] **I'm adding a new feature**
  - [x] I've updated the playground with an example use of the feature

### CODE REVIEW

**Review by epicfaace — COMMENTED** [review]
Thanks for this change! Do you mind also adding a documentation page called "3.x upgrade guide" that describes the nature of this change, and also what people should do (what polyfills they should install) if they were depending on `@babel/runtime-corejs2` that used to be bundled with `@rjsf/core`?

File: docs/3.x upgrade guide.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+### Bring your own polyfills
+
+core-js@2 has been removed from @rjsf/core. See more about [the rationale here](https://github.com/rjsf-team/react-jsonschema-form/pull/2211#issue-563700810).
+
+If you're using a framework like [Create React App](https://create-react-app.dev/docs/supported-browsers-features#supported-browsers), [Gatsby](https://www.gatsbyjs.com/docs/how-to/custom-configuration/browser-support/), or [Next.js](https://nextjs.org/docs/basic-features/supported-browsers-features), polyfills are already included there.
+
+If you were directly depending on @rjsf/core's @babel/runtime pulling in core-js@2, [@babel/preset-env](https://babeljs.io/docs/en/babel-preset-env#how-does-it-work) is probably a good first choice.
+
+From @babel/preset-envs docs
```
Comment (epicfaace) [inline docs/3.x upgrade guide.md:13]: Can we also link to the docs here?
```suggestion
From `@babel/preset-env`'s docs
```

File: docs/3.x upgrade guide.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# 3.x Upgrade Guide
+
+## BREAKING
+
+### Bring your own polyfills
+
+core-js@2 has been removed from @rjsf/core. See more about [the rationale here](https://github.com/rjsf-team/react-jsonschema-form/pull/2211#issue-563700810).
+
+If you're using a framework like [Create React App](https://create-react-app.dev/docs/supported-browsers-features#supported-browsers), [Gatsby](https://www.gatsbyjs.com/docs/how-to/custom-configuration/browser-support/), or [Next.js](https://nextjs.org/docs/basic-features/supported-browsers-features), polyfills are already included there.
```
Comment (epicfaace) [inline docs/3.x upgrade guide.md:9]:
```suggestion
If you're using a framework like [Create React App](https://create-react-app.dev/docs/supported-browsers-features#supported-browsers), [Gatsby](https://www.gatsbyjs.com/docs/how-to/custom-configuration/browser-support/), [Next.js](https://nextjs.org/docs/basic-features/supported-browsers-features), or transpiling code through something like `@babel/preset-env`, polyfills are already included there and you won't have to do anything.
```

File: docs/3.x upgrade guide.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+If you were directly depending on @rjsf/core's @babel/runtime pulling in core-js@2, [@babel/preset-env](https://babeljs.io/docs/en/babel-preset-env#how-does-it-work) is probably a good first choice.
```
Comment (epicfaace) [inline docs/3.x upgrade guide.md:11]: Just want to clarify here -- is the only solution here that we have to begin transpiling the code with `@babel/preset-env`? What if we just ran `npm install core-js` and added `import 'core-js';` -- would that also be a solution?

Comment (wegry) [reply docs/3.x upgrade guide.md]: I added a note to that effect.

**Review by epicfaace — APPROVED** [review]
Looks good! Thanks!

**Comment by epicfaace** [comment]
@wegry Can you merge master into this branch? Then we should be all ready to merge.

**Comment by wegry** [comment]
Let me give it a shot @epicfaace.

**Pull request merged**